server/services/survey.service.js

```diff
@@ -0,0 +1,60 @@
+const Path = require('path-parser');
+const { URL } = require('url');
+
+module.exports = class SurveyService {
+
+  async updateSurveys(data) {
```

Comment:
So, here's my big service method that does some work on the data sent in and then updates the database. The old code had many problems besides just being in the router.

The biggest was that even though it was an async operation (it wrote to the database), it did not use async await. This meant that if there was an error writing to the database, you would never know it. It would simply return instantly while asyncrhounously the database would be written to. I fix that by actually using async/await here, and wrapping it in a try catch block.